File: 구름 알고리즘 미션/완전 탐색/구름 RPG 2/구름 RPG 2.js
// Run by Node.js
const readline = require('readline');

(async () => {
	let rl = readline.createInterface({ input: process.stdin });
	let input = [];
	
	for await (const line of rl) {
		input.push(line);
		if(input.length === input[0]+1) rl.close();
	}
	
	function isPrime(n){
		if(n < 2) return false;
		
		for(let i = 2 ; i < parseInt(n ** 0.5) + 1 ; i++){
			if(n % i === 0) return false;
		}
		
		return true;
	}
	
	const N = +input[0];
	for(let i = 1; i <= N ; i++){
		let result = 0;
		const val = input[i];
		while(!isPrime(val - result)) result++;
		console.log(result);
	}
	
	process.exit();
})();

File: 구름 알고리즘 미션/완전 탐색/구름 RPG 2/구름 RPG 2.py
N = int(input())
A = [int(input()) for _ in range(N)]

def is_prime(n):
	for i in range(2, int(n ** 0.5) + 1):
		if not n % i:
			return False
	return True

for i in range(N):
	result = 0
	while not is_prime(A[i] - result):
		result += 1
	print(result)